Re-enable aarch64-unknown-linux-gnu target for pact_ffi

Tracking issue for work to renable aarch64-unknown-linux-gnu target for pact_ffi

Disabling commit:- https://github.com/pact-foundation/pact-reference/commit/a3025b14b8dbea1afea48ece8fa32cdd6718797d

Apparently this was an issue with Cross.

Things we can try

- Build with the latest cross
  - using an install from the repo, will give us the latest changes, this has fixed issues for me on android builds before, where I had issues in the released binaries. See [here](https://github.com/cross-rs/cross#installation) for instructions
   - 🚧  Test on GHA
     - https://github.com/YOU54F/pact-reference/actions/runs/6162735670
- Build on an arm64 machine, eliminating the need for cross
  - Test out with cirrus-cli
    - ✅ 
    - <img width="641" alt="Screenshot 2023-09-12 at 18 19 14" src="https://github.com/pact-foundation/pact-reference/assets/19932401/f2195cdb-6f2d-403a-a794-3cb9e885405e">
  - Test out with cirrus-ci
    - ✅ https://cirrus-ci.com/github/YOU54F/pact-reference/target%2Faarch64-unknown-linux-gnu

Testing branch 
  - https://github.com/YOU54F/pact-reference/tree/target/aarch64-unknown-linux-gnu

Unable to build it locally with cross on a m1 pro 

```console
rustc --version 
rustc 1.71.0 (8ede3aae2 2023-07-12)
cross --version                                         
cross 0.2.5
```

error

```console
error: linking with `aarch64-linux-gnu-gcc` failed: exit status: 1
  |
...<many paths>
/target/aarch64-unknown-linux-gnu/release/deps/libpact_verifier.so" "-Wl,--gc-sections" "-shared" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-nodefaultlibs"
  = note: /usr/lib/gcc-cross/aarch64-linux-gnu/5/../../../../aarch64-linux-gnu/bin/ld: GFp_armcap_P: undefined version: 
          /usr/lib/gcc-cross/aarch64-linux-gnu/5/../../../../aarch64-linux-gnu/bin/ld: failed to set dynamic section sizes: Bad value
          collect2: error: ld returned 1 exit status
          

error: could not compile `pact_verifier` (lib) due to previous errorc
```

--- a/rust/pact_ffi/release-linux.sh
+++ b/rust/pact_ffi/release-linux.sh
@@ -36,11 +36,10 @@ cargo build --release --target=x86_64-unknown-linux-musl
 gzip -c ../target/x86_64-unknown-linux-musl/release/libpact_ffi.a > ../target/artifacts/libpact_ffi-linux-x86_64-musl.a.gz
 openssl dgst -sha256 -r ../target/artifacts/libpact_ffi-linux-x86_64-musl.a.gz > ../target/artifacts/libpact_ffi-linux-x86_64-musl.a.gz.sha256
 
-#echo -- Build the aarch64 release artifacts --
-#rustup toolchain install 1.69
-#cargo install cross
-#rustup run 1.69 cross build --target aarch64-unknown-linux-gnu --release
-#gzip -c ../target/aarch64-unknown-linux-gnu/release/libpact_ffi.so > ../target/artifacts/libpact_ffi-linux-aarch64.so.gz
-#openssl dgst -sha256 -r ../target/artifacts/libpact_ffi-linux-aarch64.so.gz > ../target/artifacts/libpact_ffi-linux-aarch64.so.gz.sha256
-#gzip -c ../target/aarch64-unknown-linux-gnu/release/libpact_ffi.a > ../target/artifacts/libpact_ffi-linux-aarch64.a.gz
-#openssl dgst -sha256 -r ../target/artifacts/libpact_ffi-linux-aarch64.a.gz > ../target/artifacts/libpact_ffi-linux-aarch64.a.gz.sha256
+echo -- Build the aarch64 release artifacts --
+cargo install cross --git https://github.com/cross-rs/cross
+cross build --target aarch64-unknown-linux-gnu --release
+gzip -c ../target/aarch64-unknown-linux-gnu/release/libpact_ffi.so > ../target/artifacts/libpact_ffi-linux-aarch64.so.gz
+openssl dgst -sha256 -r ../target/artifacts/libpact_ffi-linux-aarch64.so.gz > ../target/artifacts/libpact_ffi-linux-aarch64.so.gz.sha256
+gzip -c ../target/aarch64-unknown-linux-gnu/release/libpact_ffi.a > ../target/artifacts/libpact_ffi-linux-aarch64.a.gz
+openssl dgst -sha256 -r ../target/artifacts/libpact_ffi-linux-aarch64.a.gz > ../target/artifacts/libpact_ffi-linux-aarch64.a.gz.sha256

